pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -146,11 +145,28 @@ func (bc *backendPoolTypeNodeIPConfig) ReconcileBackendPools(clusterName string,
 	lbBackendPoolID := bc.getBackendPoolID(lbName, bc.getLoadBalancerResourceGroup(), lbBackendPoolName)
 	vmSetName := bc.mapLoadBalancerNameToVMSet(lbName, clusterName)
 
-	for _, bp := range newBackendPools {
+	for i := len(newBackendPools) - 1; i >= 0; i-- {
+		bp := newBackendPools[i]
 		if strings.EqualFold(*bp.Name, lbBackendPoolName) {
-			klog.V(10).Infof("bc.ReconcileBackendPools for service (%s)(%t): lb backendpool - found wanted backendpool. not adding anything", serviceName, wantLb)
+			klog.V(10).Infof("bc.ReconcileBackendPools for service (%s): lb backendpool - found wanted backendpool. not adding anything", serviceName)
 			foundBackendPool = true
 
+			// If the LB backend pool type is configured from nodeIP or podIP
+			// to nodeIPConfiguration, we need to decouple the VM NICs from the LB
+			// before attaching nodeIPs/podIPs to the LB backend pool.
+			if bp.BackendAddressPoolPropertiesFormat != nil &&
+				(bp.BackendIPConfigurations == nil || len(*bp.BackendIPConfigurations) == 0) {
```

Comment:
I determine the backend pool type by checking if there are backend IP configs in the pool. However, for those agent pools with zero instance, it may bring up an extra api call. Is there a better way to check the type? I noticed there is `Type` in the backend pool body, but I am not sure if that would be the type of IP/IPConfig based backend pool. And I also don't know the possible values of the `Type`.

Reply:
"type": "Microsoft.Network/loadBalancers/backendAddressPools"
It looks we cannot tell the difference between these two kinds of backend pools by `type`. It's always the same value.

Reply:
yes, it's only possible from either networkInterfaceIPConfiguration or ipAddress on existing backend pool (need at least one backend pool though).